src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -69,15 +72,136 @@ protected bool CompileVersionBubbleGenericsIntoCurrentModule(MethodDesc method)
             return true;
         }
 
-        /// <summary>
-        /// If true, the type is fully contained in the current compilation group.
-        /// </summary>
-        public override bool ContainsTypeLayout(TypeDesc type)
+        public CompilationUnitSet TypeLayoutCompilationUnits(TypeDesc type)
         {
-            return _containsTypeLayoutCache.GetOrAdd(type, ContainsTypeLayoutUncached);
+            return _layoutCompilationUnits.GetOrAdd(type, TypeLayoutCompilationUnitsUncached);
         }
 
-        private bool ContainsTypeLayoutUncached(TypeDesc type)
+        // Compilation Unit Index is the compilation unit of a given module. If the compilation unit
+        // is unknown the module will be given an independent index from other modules, but 
+        // IsCompilationUnitIndexExact will return false for that index. All compilation unit indices 
+        // are >= 2, to allow for 0 to be a sentinel value.
+        private int ModuleToCompilationUnitIndex(ModuleDesc nonEcmaModule)
+        {
+            EcmaModule module = (EcmaModule)nonEcmaModule;
+            if (IsModuleInCompilationGroup(module))
+                return 2;
+
+            if (!VersionsWithModule(module))
+                return 3;
+            
+            lock (_moduleCompilationUnits)
+            {
+                if (!_moduleCompilationUnits.TryGetValue(module, out int compilationUnit))
+                {
+                    compilationUnit = ++_maxCompilationUnitsKnown;
+                    _moduleCompilationUnits.Add(module, compilationUnit);
+                }
+
+                return compilationUnit;
+            }
+        }
+
+        // Indicate 
```

Comment:
What does this comment mean?

Reply:
I fell asleep before finishing writing it... update on the way along with use of an enum instead of an int.